### PR TITLE
feat: notifications onboarding wizard

### PR DESCRIPTION
**Pull request by Jonathansoufer**
## **Description**

This PR is based on [9263](https://github.com/MetaMask/metamask-mobile/pull/9263), reverted due an e2e failure, which is fixed on this PR. 

## **Related issues**

Fixes:

## **Manual testing steps**

Install a fresh app and accept the tour at beginning.

### **After**

https://github.com/MetaMask/metamask-mobile/assets/44679989/f17e1a1d-43a3-4234-840e-9d93731f9370

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/OnboardingWizard/Step2/index.js [hunk 1]
```diff
@@ -27,95 +28,59 @@ const styles = StyleSheet.create({
   },
 });
 
-class Step2 extends PureComponent {
```
Comment (tommasini) [inline app/components/UI/OnboardingWizard/Step2/index.js:30]: Love what you did here! If you see the possibility, converting it to typescript would be amazing!

File: app/components/UI/OnboardingWizard/Step7/index.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
+import React, { useEffect, useState } from 'react';
```
Comment (tommasini) [inline app/components/UI/OnboardingWizard/Step7/index.js:1]: Do you think we could convert this new component to typescript?

**Review by tommasini — COMMENTED** [review]
Amazing job here! ❤️
Just left comments for we convert to typescript the new step 7 component and if possible the one that was converted to functional component to typescript as well. That will help US continue growing the typescript coverage of the app 🙏🏼

File: app/components/UI/OnboardingWizard/Step2/index.js (hunk 1 above)
Comment (Jonathansoufer) [reply app/components/UI/OnboardingWizard/Step2/index.js:30]: Surething!

File: app/components/UI/OnboardingWizard/Step7/index.js (hunk 2 above)
Comment (Jonathansoufer) [reply app/components/UI/OnboardingWizard/Step7/index.js]: Surething!

**Review by brianacnguyen — DISMISSED** [review]

**Review by legobeat — DISMISSED** [review]
Unit tests are failing: https://github.com/MetaMask/metamask-mobile/actions/runs/8900659423/job/24442767307?pr=9392

**Comment by NicolasMassart** [comment]
On iOS, the tour bubbles are not correctly placed relatively to the point they show, it may even hide the buttons a bit.
|2/6|3/6|4/6|5/6|6/6|
|-|-|-|-|-|
||||||

**Review by NicolasMassart — COMMENTED** [review]
On iOS, the tour bubbles are not correctly placed relatively to the point they show, it may even hide the buttons a bit.
|2/6|3/6|4/6|5/6|6/6|
|-|-|-|-|-|
||||||

**Comment by Jonathansoufer** [comment]
> On iOS, the tour bubbles are not correctly placed relatively to the point they show, it may even hide the buttons a bit.
> 
> 2/6	3/6	4/6	5/6	6/6
> 				

Addressed! 

https://github.com/MetaMask/metamask-mobile/assets/44679989/8dde36e5-a0f7-4dbd-840c-d870fb4b4117

**Review by NicolasMassart — COMMENTED** [review]
The Positions look better now even if I steel have a small missalignment for notifications step 3/6 (iOS 17.4 iPhone 13 pro simulator)
image
Is there a Figma I could check to see if it's expected? I haven't found any in the original PR as it's not linked to any issue (it should).

As indicated by Sonar, you will also have ot fix the duplication to pass the gate and merge.
I suggest that you create a common component from all steps by picking the common code parts and then only have the variations on each specific step.

**Comment by Jonathansoufer** [comment]
> The Positions look better now even if I steel have a small missalignment for notifications step 3/6 (iOS 17.4 iPhone 13 pro simulator) image Is there a Figma I could check to see if it's expected? I haven't found any in the original PR as it's not linked to any issue (it should).
> 
> As indicated by Sonar, you will also have ot fix the duplication to pass the gate and merge. I suggest that you create a common component from all steps by picking the common code parts and then only have the variations on each specific step.

Fixed.

Screenshot 2024-05-07 at 12 58 11

**Review by NicolasMassart — APPROVED** [review]
Looks good to me. Nice work!

**Pull request merged**